src/commands.rs

```diff
@@ -146,15 +146,78 @@ fn run_server_process() -> Result<ServerStartup> {
 #[cfg(not(windows))]
 fn daemonize() -> Result<()> {
     use daemonize::Daemonize;
-    if match env::var("SCCACHE_NO_DAEMON") {
-            Ok(val) => val == "1",
-            Err(_) => false,
-    } {
-        Ok(())
-    } else {
-        Daemonize::new().start().chain_err(|| {
-            "failed to daemonize"
-        })
+    use libc;
+    use std::mem;
+
+    match env::var("SCCACHE_NO_DAEMON") {
+        Ok(ref val) if val == "1" => {}
+        _ => {
+            Daemonize::new().start().chain_err(|| {
+                "failed to daemonize"
+            })?;
+        }
+    }
+
+    static mut PREV_SIGSEGV: *mut libc::sigaction = 0 as *mut _;
+    static mut PREV_SIGBUS: *mut libc::sigaction = 0 as *mut _;
+    static mut PREV_SIGILL: *mut libc::sigaction = 0 as *mut _;
+
+    // We don't have a parent process any more once we've reached this point,
+    // which means that no one's probably listening for our exit status.
+    // In order to assist with debugging crashes of the server we configure our
+    // rlimit to allow core dumps and we also install a signal handler for
+    // segfaults which at least prints out what just happened.
+    unsafe {
+        let rlim = libc::rlimit {
+            rlim_cur: libc::RLIM_INFINITY,
+            rlim_max: libc::RLIM_INFINITY,
+        };
+        libc::setrlimit(libc::RLIMIT_CORE, &rlim);
```

Comment:
This worries me a little bit, in that for non-ephemeral build environments (like OS X) we could pile up coredumps without noticing. Could we put just the call to `setrlimit` behind an environment variable?